renderdoc/data/glsl/gles_texsample.h

```diff
@@ -184,11 +203,14 @@ vec4 SampleTextureFloat4(int type, vec2 pos, float slice, int mipLevel, int samp
   }
   else if(type == RESTYPE_TEXBUFFER)
   {
-    col = texelFetch(texBuffer, int(pos.x * texRes.x));
 #ifdef OPENGL_ES
     // This hack is needed for an Android device to let the compiler optimize out the texBuffer,
     // because otherwise due to some compiler bug the RESTYPE_TEX2D case won't work normally.
     col = vec4(1.0f, 1.0f, 1.0f, 1.0f);
+#elif defined(GLES_texture_buffer)
+    col = texelFetch(texBuffer, int(pos.x * texRes.x));
+#else
+    col = vec4(0.0f, 0.0f, 0.0f, 0.0f);
```

Comment:
I think it might be better to separate this out into two #ifs. That way there is one #if which controls checking for the extension and either doing the sample or returning black - just like the other cases. Then there's a separate #if after that with the hack to let the compiler optimize out the code. Then it's easier for someone to see what code should be removed when the hack isn't necessary (I don't know if that will ever happen). Something like this:

```cpp
#if defined(GLES_texture_buffer)
    col = texelFetch(texBuffer, int(pos.x * texRes.x));
#else
    col = vec4(0.0f, 0.0f, 0.0f, 0.0f);
#endif

#ifdef OPENGL_ES
    // This hack is needed for an Android device to let the compiler optimize out the texBuffer,
    // because otherwise due to some compiler bug the RESTYPE_TEX2D case won't work normally.	
    col = vec4(1.0f, 1.0f, 1.0f, 1.0f);
#endif
```

@pantosj do you have any preference?